src/distances.jl

```diff
@@ -34,3 +26,23 @@ with `T` of appropriate type for members of the set.
 function distance_to_set end
 
 distance_to_set(::AbstractDistance, v, s) = distance_to_set(DefaultDistance(), v, s)
+
+"""
+    EpigraphViolationDistance
+
+Distance used when `v ∈ s` can be expressed as a single scalar function `f(v) ≤ 0`.
+The distance is expressed as `max(f(v), 0)`.
+"""
+struct EpigraphViolationDistance <: AbstractDistance end
+
+"""
+    NormedEpigraphDistance{p}
+
+Distance used when `v ∈ s` can be expressed as a several functions `f_i(v) ≤ 0 ∀ i ∈ I`.
```

Comment:
I would maybe rephrase, e.g. `as several functions` , or `as an intersection of different level sets`

Reply:
yes thanks, intersection of level sets is the best way to describe it IMO